src/transport.rs

```diff
@@ -2,6 +2,15 @@ use std::io;
 use std::io::{Read, Write};
 use std::net;
 use std::net::TcpStream;
+#[cfg(feature = "ssl")]
+use openssl::ssl::{SslStream, SslConnector};
+
+pub  trait  CDRSTransport: Sized+Read+Write+Send+Sync{
+
+    fn try_clone(&self) -> io::Result<Self>;
+    fn close(&mut self, close: net::Shutdown) -> io::Result<()>;
+}
+
 
 #[derive(Debug)]
 pub struct Transport {
```

Comment:
`TransportTcp` might be a little counterintuitive; as tls do communicate over Tcp.
I renamed to  `TransportPlain` and `TransportTls`